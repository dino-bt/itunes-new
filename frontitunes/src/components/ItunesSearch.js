import React, { useState } from "react";
//import ItemsCard from "./ItemsCard";
import { FcSearch } from "react-icons/fc";
import Card from "react-bootstrap/Card";
import { MdFavorite } from "react-icons/md";
import FavouriteCard from "./FavouriteCard";

const ItunesSearch = () => {
  //we create two use states where we will capture the information of the artist and the entity so that we can later use it to put it into the
  //from the api
  //we first get the the name of the artist the term and then api's and then we get the entity and set the entity
  //back into the api which we get a request from the server to add the information
  //we get the information back and set the information into items.
  //we will then create a state when the user clicks on favourites and then in favourites map through that to get our result
  const [items, setItems] = useState([]);
  const [term, setTerm] = useState("");
  const [option, setOption] = useState("");
  const [fav, setFav] = useState([]);
  //fetch the api from the server
  const searchItunes = async (e) => {
    e.preventDefault();

    fetch("/search", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        term: term,
        option: option,
      }),
    })
      .then((res) => res.json())
      .then((data) => {
        setItems(data.results);
      });

    //const res = await fetch(`/search/${term}/${option}`);
    //const data = await res.json();
    //console.log(data)

    //setItems(data.results);
  };
  //here we will setFavourites by adding the new item into a fav array
  function setFavourites(item) {
    setFav([...fav, item]);
    alert("Your Item is now in favourites");
    console.log(fav);
  }

  return (
    <div className="searchDiv">
      <h3 id="searchId">SEARCH YOUR FAVOURITE ARTIST</h3>
      <hr />
      <div className="searchBox">
        <form onSubmit={searchItunes}>
          <input
            className="artistName"
            type="text"
            name="term"
            placeholder="Artist Name"
            value={term}
            onChange={(e) => setTerm(e.target.value)}
          />
          <br />
          <br />

          <button className="searchButton">
            SEARCH <FcSearch />
          </button>

          <br />

          <select
            className="select"
            value={option}
            onChange={(e) => setOption(e.target.value)}
            placeholder="media"
          >
            <option value="all">All</option>
            <option value="movie">Movie</option>
            <option value="podcast">Podcast</option>
            <option value="tvShow">TV Show</option>
            <option value="audiobook">Audio Book</option>

            <option value="music">Songs</option>

            <option value="ebook">E-Book</option>
          </select>
          <br />
        </form>
      </div>

      <Card>
        {items.map((item, idz) => (
          <div className="display-card-body" key={idz}>
            <Card.Link>
              <Card.Img src={item.artworkUrl100} className="card-img" />
            </Card.Link>
            <Card.Body>
              <Card.Title>
                <strong> {item.trackName} </strong>
              </Card.Title>
              <Card.Subtitle className="mb-2 text-muted">
                {" "}
                {item.collectionId}{" "}
              </Card.Subtitle>
              <Card.Subtitle className="mb-2 card-type">
                {" "}
                {item.kind}{" "}
              </Card.Subtitle>
              <Card.Text> {item.artistName} </Card.Text>
            </Card.Body>
            <button
              className="favButton"
              id="addFavBtn"
              onClick={() => setFavourites(item)}
            >
              <MdFavorite /> FAVOURITE
            </button>
          </div>
        ))}

        <FavouriteCard favourite={fav} setFav={setFav} />
      </Card>
    </div>
  );
};

//we map through the state(items) to get the id the kind the name the artist and the image
export default ItunesSearch;
