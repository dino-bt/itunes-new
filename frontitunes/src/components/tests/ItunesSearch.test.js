import React from "react";
import renderer from "react-test-renderer";


import ItunesSearch  from "../ItunesSearch";

it("tests the ItuneSearch component", () => {
  const tree = renderer.create(<ItunesSearch/>).toJSON();
  expect(tree).toMatchSnapshot();
});