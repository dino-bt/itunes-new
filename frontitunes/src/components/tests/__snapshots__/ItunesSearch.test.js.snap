// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the Header component 1`] = `
<div
  className="searchDiv"
>
  <h3
    id="searchId"
  >
    SEARCH YOUR FAVOURITE ARTIST
  </h3>
  <hr />
  <div
    className="searchBox"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="artistName"
        name="term"
        onChange={[Function]}
        placeholder="Artist Name"
        type="text"
        value=""
      />
      <br />
      <br />
      <button
        className="searchButton"
      >
        SEARCH 
        <svg
          enableBackground="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#616161"
          >
            <rect
              height="17"
              transform="matrix(.707 -.707 .707 .707 -15.154 36.586)"
              width="4"
              x="34.6"
              y="28.1"
            />
            <circle
              cx="20"
              cy="20"
              r="16"
            />
          </g>
          <rect
            fill="#37474F"
            height="12.3"
            transform="matrix(.707 -.707 .707 .707 -15.839 38.239)"
            width="4"
            x="36.2"
            y="32.1"
          />
          <circle
            cx="20"
            cy="20"
            fill="#64B5F6"
            r="13"
          />
          <path
            d="M26.9,14.2c-1.7-2-4.2-3.2-6.9-3.2s-5.2,1.2-6.9,3.2c-0.4,0.4-0.3,1.1,0.1,1.4c0.4,0.4,1.1,0.3,1.4-0.1 C16,13.9,17.9,13,20,13s4,0.9,5.4,2.5c0.2,0.2,0.5,0.4,0.8,0.4c0.2,0,0.5-0.1,0.6-0.2C27.2,15.3,27.2,14.6,26.9,14.2z"
            fill="#BBDEFB"
          />
        </svg>
         
      </button>
      <br />
      <select
        className="select"
        onChange={[Function]}
        placeholder="Media"
        value=""
      >
        <option
          value="select"
        >
          Select
        </option>
        <option
          value="all"
        >
          All
        </option>
        <option
          value="movie"
        >
          Movie
        </option>
        <option
          value="podcast"
        >
          Podcast
        </option>
        <option
          value="tvShow"
        >
          TV Show
        </option>
        <option
          value="audiobook"
        >
          Audio Book
        </option>
        <option
          value="shortfilm"
        >
          Short Film
        </option>
        <option
          value="music"
        >
          Songs
        </option>
        <option
          value="ebook"
        >
          E-Book
        </option>
      </select>
      <br />
    </form>
  </div>
  <div
    className="card"
  />
  <br />
  <div
    className="favouriteBlock"
  >
    <div
      className="insideFavourite"
    >
      <h1
        className="hFavourites"
      >
        FAVOURITES
      </h1>
      <div
        className="card"
        id="favouritesId"
      />
    </div>
  </div>
</div>
`;

exports[`tests the I component 1`] = `
<div
  className="searchDiv"
>
  <h3
    id="searchId"
  >
    SEARCH YOUR FAVOURITE ARTIST
  </h3>
  <hr />
  <div
    className="searchBox"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="artistName"
        name="term"
        onChange={[Function]}
        placeholder="Artist Name"
        type="text"
        value=""
      />
      <br />
      <br />
      <button
        className="searchButton"
      >
        SEARCH 
        <svg
          enableBackground="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#616161"
          >
            <rect
              height="17"
              transform="matrix(.707 -.707 .707 .707 -15.154 36.586)"
              width="4"
              x="34.6"
              y="28.1"
            />
            <circle
              cx="20"
              cy="20"
              r="16"
            />
          </g>
          <rect
            fill="#37474F"
            height="12.3"
            transform="matrix(.707 -.707 .707 .707 -15.839 38.239)"
            width="4"
            x="36.2"
            y="32.1"
          />
          <circle
            cx="20"
            cy="20"
            fill="#64B5F6"
            r="13"
          />
          <path
            d="M26.9,14.2c-1.7-2-4.2-3.2-6.9-3.2s-5.2,1.2-6.9,3.2c-0.4,0.4-0.3,1.1,0.1,1.4c0.4,0.4,1.1,0.3,1.4-0.1 C16,13.9,17.9,13,20,13s4,0.9,5.4,2.5c0.2,0.2,0.5,0.4,0.8,0.4c0.2,0,0.5-0.1,0.6-0.2C27.2,15.3,27.2,14.6,26.9,14.2z"
            fill="#BBDEFB"
          />
        </svg>
         
      </button>
      <br />
      <select
        className="select"
        onChange={[Function]}
        placeholder="Media"
        value=""
      >
        <option
          value="select"
        >
          Select
        </option>
        <option
          value="all"
        >
          All
        </option>
        <option
          value="movie"
        >
          Movie
        </option>
        <option
          value="podcast"
        >
          Podcast
        </option>
        <option
          value="tvShow"
        >
          TV Show
        </option>
        <option
          value="audiobook"
        >
          Audio Book
        </option>
        <option
          value="shortfilm"
        >
          Short Film
        </option>
        <option
          value="music"
        >
          Songs
        </option>
        <option
          value="ebook"
        >
          E-Book
        </option>
      </select>
      <br />
    </form>
  </div>
  <div
    className="card"
  />
  <br />
</div>
`;

exports[`tests the ItuneSearch component 1`] = `
<div
  className="searchDiv"
>
  <h3
    id="searchId"
  >
    SEARCH YOUR FAVOURITE ARTIST
  </h3>
  <hr />
  <div
    className="searchBox"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="artistName"
        name="term"
        onChange={[Function]}
        placeholder="Artist Name"
        type="text"
        value=""
      />
      <br />
      <br />
      <button
        className="searchButton"
      >
        SEARCH 
        <svg
          enableBackground="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#616161"
          >
            <rect
              height="17"
              transform="matrix(.707 -.707 .707 .707 -15.154 36.586)"
              width="4"
              x="34.6"
              y="28.1"
            />
            <circle
              cx="20"
              cy="20"
              r="16"
            />
          </g>
          <rect
            fill="#37474F"
            height="12.3"
            transform="matrix(.707 -.707 .707 .707 -15.839 38.239)"
            width="4"
            x="36.2"
            y="32.1"
          />
          <circle
            cx="20"
            cy="20"
            fill="#64B5F6"
            r="13"
          />
          <path
            d="M26.9,14.2c-1.7-2-4.2-3.2-6.9-3.2s-5.2,1.2-6.9,3.2c-0.4,0.4-0.3,1.1,0.1,1.4c0.4,0.4,1.1,0.3,1.4-0.1 C16,13.9,17.9,13,20,13s4,0.9,5.4,2.5c0.2,0.2,0.5,0.4,0.8,0.4c0.2,0,0.5-0.1,0.6-0.2C27.2,15.3,27.2,14.6,26.9,14.2z"
            fill="#BBDEFB"
          />
        </svg>
         
      </button>
      <br />
      <select
        className="select"
        onChange={[Function]}
        placeholder="Media"
        value=""
      >
        <option
          value="select"
        >
          Select
        </option>
        <option
          value="all"
        >
          All
        </option>
        <option
          value="movie"
        >
          Movie
        </option>
        <option
          value="podcast"
        >
          Podcast
        </option>
        <option
          value="tvShow"
        >
          TV Show
        </option>
        <option
          value="audiobook"
        >
          Audio Book
        </option>
        <option
          value="shortfilm"
        >
          Short Film
        </option>
        <option
          value="music"
        >
          Songs
        </option>
        <option
          value="ebook"
        >
          E-Book
        </option>
      </select>
      <br />
    </form>
  </div>
  <div
    className="card"
  >
    <div
      className="favouriteBlock"
    >
      <div
        className="insideFavourite"
      >
        <h1
          className="hFavourites"
        >
          FAVOURITES
        </h1>
        <div
          className="card"
          id="favouritesId"
        />
      </div>
    </div>
  </div>
  <br />
</div>
`;
